croncat/src/rpc/signer.rs

```diff
@@ -17,150 +19,134 @@ use cw_rules_core::msg::QueryConstruct;
 use cw_rules_core::types::CroncatQuery;
 use futures_util::Future;
 use serde::de::DeserializeOwned;
-use tendermint_rpc::endpoint::broadcast::tx_commit::TxResult;
+use serde::Serialize;
 use tokio::time::timeout;
 
-use crate::client::full_client::CosmosFullClient;
-use crate::client::QueryBank;
 use crate::config::ChainConfig;
 use crate::errors::{eyre, Report};
 
-#[derive(Clone)]
-pub struct GrpcSigner {
-    client: CosmosFullClient,
+use super::RpcClient;
+
+#[derive(Clone, Debug)]
+pub struct Signer {
+    rpc_client: RpcClient,
     pub manager: String,
     pub account_id: AccountId,
 }
 
-impl GrpcSigner {
+impl Signer {
     pub async fn new(
         rpc_url: String,
-        grpc_url: String,
-        chain_info: ChainInfo,
+        cfg: ChainConfig,
         manager: String,
         key: bip32::XPrv,
-        gas_prices: f32,
-        gas_adjustment: f32,
+        mnemonic: String,
     ) -> Result<Self, Report> {
         // TODO: How should we handle this? Is the hack okay?
         // Quick hack to add https:// to the url if it is missing
-        let grpc_url = if !grpc_url.starts_with("https://") {
-            format!("https://{}", grpc_url)
-        } else {
-            grpc_url
-        };
         let rpc_url = if !rpc_url.starts_with("https://") {
             format!("https://{}", rpc_url)
         } else {
             rpc_url
         };
 
-        let client = timeout(
-            Duration::from_secs(10),
-            CosmosFullClient::new(
-                rpc_url,
-                grpc_url,
-                chain_info,
-                key,
-                gas_prices,
-                gas_adjustment,
-            ),
-        )
-        .await??;
-        let account_id = client
-            .key()
+        // Get the account id from the key.
+        let signing_key: SigningKey = key.into();
+        let account_id = signing_key
             .public_key()
-            .account_id(&client.chain_info.bech32_prefix)?;
+            .account_id(&cfg.info.bech32_prefix)?;
+
+        // Create a new RPC client
+        let mut rpc_client = RpcClient::new(&cfg, rpc_url.as_str())?;
+        // TODO: This is a hack to get around the fact that cosm-tome doesn't
+        // let us pass an xprv.
+        rpc_client.set_mnemonic(mnemonic);
+        rpc_client.set_denom(&cfg.info.fees.fee_tokens[0].denom);
```

Comment:
in the future, we can also use a config variable right? 
Chain config has a higher level one as well above fee_tokens called `base`